Framework/Core/include/Framework/ASoA.h

```diff
@@ -1988,6 +1988,63 @@ std::tuple<typename Cs::type...> getRowData(arrow::Table* table, T rowIterator,
 {
   return std::make_tuple(getSingleRowData<T, Cs>(table, rowIterator, ci, ai, globalIndex)...);
 }
+
+template <typename R, typename T, typename C>
+R getColumnValue(const T& rowIterator)
+{
+  return static_cast<R>(static_cast<C>(rowIterator).get());
```

Comment:
Even in my use case, columns have different types, and this cast to R is needed to make a common "ColumnGetterFunction" type. The first static cast chooses a correct `get` function from `rowIterator`. Maybe I could skip this R cast explicitly, and it would be done automatically (but I need to test if it would work like that). Would it look tidier, then?

Reply:
I think my point is more about the C cast. I would assume that if you can do `static_cast<R>(static_cast<C>(x))`, you can also do `static_cast<R>(x)` directly, no?

Reply:
R is short for common return type, which can be ex. float. This C cast is only needed to use the `get()` member method of the dynamic/persistent column, and then it is called; its return value is then cast to R type. So I can't cast it directly, because I do not know what I would cast

Reply:
Ok, understood the usecase. Shouldn't R be derivable from C?

Reply:
As is my first comment, R is needed for a common return type, to be able to store pointer to function obtained in runtime. In compile time we do not know which column label (so also column::type) would match columnLabel from JSON. In ML use case we would use only floats, so this cast to common return type is desired.

Reply:
Ok, I understand. No further comments on this for now.